src/vcpkg/archives.cpp

```diff
@@ -177,6 +187,31 @@ namespace
 
 namespace vcpkg
 {
+    void extract_self_extracting_7z(const VcpkgPaths& paths, const Path& archive, const Path& to_path)
+    {
+        // static_cast due to Warning C4309
+        constexpr static const char header_7z[] = {
+            '7', 'z', static_cast<char>(0xbc), static_cast<char>(0xaf), 0x27, 0x1c};
+
+        const Path stem(archive.stem());
+        const auto subext = stem.extension();
+        Checks::check_exit(VCPKG_LINE_INFO,
+                           Strings::case_insensitive_ascii_equals(subext, ".7z"),
+                           "Unable to extract 7z archive from Installer %s. Missing '.7z.exe' extension.",
+                           archive);
+
+        Filesystem& fs = paths.get_filesystem();
+        auto contents = fs.read_contents(archive, VCPKG_LINE_INFO);
+        auto pos = Strings::search(contents, header_7z);
+        Checks::check_exit(VCPKG_LINE_INFO,
+                           pos != StringView(contents).end(),
+                           "Unable to extract 7z archive from Installer %s. Unable to find 7z header.",
+                           archive);
+        auto distance = std::distance(StringView(contents).begin(), pos);
+        contents = contents.substr(distance);
```

Comment:
```suggestion
        auto 7z_contents = StringView{contents.data(), pos);
```

Reply:
This throws:  `error C2059: Syntaxfehler: "user-defined literal"` for some reason.

Reply:
ah because you cannot start variables with numbers.

Reply:
the string view does not help here since `fs.write_contents` does not know anything about string_views.